docs/proposals/thin-snapshots-restore.md

```diff
@@ -0,0 +1,232 @@
+* `lvmd` creates an LVM logical volume as requested.
+* `lvmd` activates the thin logical volume with `Read-Write` permissions.
+* `topolvm-node` updates the status of LogicalVolume.
+* `topolvm-controller` finds the updated status of LogicalVolume.
+* `topolvm-controller` sends the success (or failure) to `provisioner` sidecar.
+
+
+## LogicalVolume CR for Restore operation
+
+```yaml
+apiVersion: topolvm.cybozu.com/v1
+kind: LogicalVolume
+metadata:
+  name: pvc-42041aee-79a2-4184-91aa-5e1df6068b9f # restore pv name
+  annotations:
+spec:
+  deviceClass: <string>
+  name: pvc-42041aee-79a2-4184-91aa-5e1df6068b9f
+  nodeName: 192.168.26.40
+  # 'snapshot' is an optional field, that needs to specified only
+  # in case of snapshot logical volume creation.
+  snapshot:
+    # 'type' specifies if the snapshot will be thick or thin provisioned.
+    type: thin
+    # 'dataSource' specifies the volumeID of the parent logical volume.
+    dataSource: <string>
+    # 'accessType' specifies how the user intents to consume the
+    # thin-snapshot logical volume.
+    # If the thin-snap lv is to be provided to user as a snapshot,
+    # the accessType can be Read-Only.
+    # On the other hand, if the lv is to be given as a persistent volume (restore),
+    # the accessType should be set to allow Read-Write operations.
+    accessType: rw
+status:
+  volumeID: <Store the volume id of the lvm snapshot as same as existing LV.>
+```
+
+## LV Operations
+
+* Snapshot creation from a thinly-provisioned volume:
+    ```bash
+    $ lvcreate -s --name thinsnap VG0/thinvolume
+    ```
+
+* Activation of a logical volume
+
+  A thinly-provisioned snapshot will be activated as Read-Only for snapshots and Read-Write for Restore scenarios.
+
+  For e.g: Activating a logical volume for Read-Write operations:
+
+  ```bash
+  $ lvchange -ay -K VG0/thinsnap
+  ```
+
+* Restoring a Snapshot
+
+
+  ```bash
+  $ lvcreate -s --name thin-restored VG0/thinsnap
+  ```
+
+## VolumeSnapshotClass and VolumeSnapshot
+
+### VolumeSnapshotClass
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: topolvm-snapclass
+driver: topolvm.cybozu.com
+deletionPolicy: Delete
+parameters:
+  # This field is for selecting the mode of restoring volume. It must be `local`
+  # for snapshots intended to be created on the same node.
+  "topolvm.cybozu.com/restore-mode": "local"
+```
+
+### VolumeSnapshot
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: new-snapshot
+spec:
+  volumeSnapshotClassName: topolvm-snapclass
+  source:
+    persistentVolumeClaimName: snapshot-pvc
+```
+
+## Deployment Changes
+
+* Add `external-snapshotter` sidecar.
+
+* Perform `snapshot-controller` deployment.
+
+## Metrics and Monitoring
+
+Should be already covered during adding support for `thin` logical volumes. See [thin-provisioning proposal doc](https://github.com/topolvm/topolvm/pull/442).
```

Comment:
PR #442 has been merged, please change the URL.
```suggestion
Should be already covered during adding support for `thin` logical volumes. See [thin-provisioning proposal doc](docs/proposals/thin-volumes.md).
```

Reply:
Thanks, updated :+1: